app/scripts/controllers/pod.js

```diff
@@ -234,6 +236,10 @@ angular.module('openshiftConsole')
               setContainerVars();
 
               updateTerminals($scope.containerTerminals);
+
+              if ($scope.containersRunning(pod.status.containerStatuses) > 0 ) {
```

Comment:
you need this logic after the makeTerminals call here https://github.com/openshift/origin-web-console/pull/342/files#diff-7f99f2d99896dd6aa2719df2a46a2acfR224

Otherwise you could get a flash of the "No containers running" message between the initial GET succeeding which triggers loaded=true and the list call that watch creates which will then trigger the noContainersYet = false to show the terminal.